core/tests/test_views.py

```diff
@@ -243,13 +226,15 @@ def test_post_user_avatar_display(client: Client, community: Community, user_wit
 
 
 @pytest.mark.django_db()
-def test_post_user_without_avatar(client: Client, community: Community, user: User, default_avatar_url: str) -> None:
```

Comment:
Why this change? Only user to another_user? Doesn't seem needed, no? If not needed, please revert.

Reply:
Second question is why are you changing tests for views if you are working on API? Please explain or revert.

Reply:
Author has been added to community fixture.

community fixture before:
```python
@pytest.fixture()
def community() -> Community:
    return Community.objects.create(name="Test Community", is_active=True)
```

Comunity fixture now below. Notice: `user` fixture, `author=user`
```python
@pytest.fixture()
def community(user: User) -> Generator[Community, None, None]:
    # community is restricted by default
    return Community.objects.create(author=user, name="Test Community")
 ```
.
.
Community fixture uses user fixture. Because of that for example here: User returned by `user` fixture and `community.user` are in conflict. user fixture is called both in this test and in Community fixture.
Result is `django.db.utils.IntegrityError: UNIQUE constraint failed: users_user.nickname`

Still I think author in Community is a must. My solution was to introduce `another_user` fixture.


```python
@pytest.fixture()
def another_user(reusable_password: Callable[[], str]) -> User:
    """Use when user different than community author is needed."""
    password = reusable_password()
    user = User.objects.create_user(email="other_user@example.com", nickname="other_user", password=password)
    user.plain_password = password
    return user
```

Reply:
TLDR; Below calls user fixture twice (user fixture is requested by community fixture).
```python
def test_post_user_without_avatar(client: Client, community: Community, user: User, default_avatar_url: str) -> None:
```

Reply:
> Second question is why are you changing tests for views if you are working on API? Please explain or revert.

This is on purpuse. Consequence of slight fixture refactor.

Reply:
Memo to self: approved.